searx/engines/devicons.py

```diff
@@ -0,0 +1,63 @@
+# SPDX-License-Identifier: AGPL-3.0-or-later
+"""Devicons (icons)"""
+
+import typing as t
+
+from searx.result_types import EngineResults
+
+if t.TYPE_CHECKING:
+    from extended_types import SXNG_Response
+    from search.processors.online import OnlineParams
+
+
+about = {
+    "website": "https://devicon.dev/",
+    "wikidata_id": None,
+    "official_api_documentation": None,
+    "use_official_api": True,
+    "results": "JSON",
+}
+
+cdn_base_url = "https://cdn.jsdelivr.net/gh/devicons/devicon@latest"
+categories = ["images", "icons"]
+
+
+def request(query: str, params: "OnlineParams"):
+    params["url"] = f"{cdn_base_url}/devicon.json"
+    params['query'] = query
+    return params
+
+
+def response(resp: "SXNG_Response") -> EngineResults:
+    res = EngineResults()
+    query_parts = resp.search_params["query"].lower().split(" ")
+
+    def is_result_match(result: dict[str, t.Any]) -> bool:
+        for part in query_parts:
+            if part in result["name"]:
+                return True
+
+            for tag in result["altnames"] + result["tags"]:
+                if part in tag:
+                    return True
+
+        return False
+
+    filtered_results = filter(is_result_match, resp.json())
+    for result in filtered_results:
+        for image_type in result["versions"]["svg"]:
+            img_src = f"{cdn_base_url}/icons/{result['name']}/{result['name']}-{image_type}.svg"
+            res.append(
+                res.types.LegacyResult(
```

Comment:
`EngineResults` is a inheritance of list and has therefore a `append` method, but we should use the dedicated method `res.add` .. 

https://github.com/searxng/searxng/blob/7a1b959646c45a81d3495148b1fa6c2da585eb59/searx/result_types/__init__.py#L65-L67

Reply:
Ah, I see. Wouldn't it make sense to add some deprecation warning to `EngineResults.append` then? I.e. 

```python
 def add(self, result: Result | LegacyResult): 
     """Add a :py:`Result` item to the result list.""" 
     super().append(result) 

 def append(self, value: Any):
    logger.debug("DeprecationWarning: Calling EngineResults.append directly is deprecated, please use EngineResults.add")
    self.add(result)
```